src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -316,7 +316,7 @@ public function request(string $method, string $url, array $options = []): Respo
         }
 
         foreach ($curlopts as $opt => $value) {
-            if (\CURLOPT_INFILESIZE === $opt && $value >= 1 << 31) {
+            if (\CURLOPT_INFILESIZE === $opt && $value >= 1 << 31 && \PHP_INT_SIZE === 8) {
```

Comment:
The `\PHP_INT_SIZE === 8` check should be the first one, as it will be resolved at compile time by OPCache (potentially skipping the whole condition if it ends up being `false`)